internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -0,0 +1,372 @@
+			timer.SleepForInterval()
+			continue
+		}
+		c.loggingClient.Info("found Consul leader to bootstrap ACL")
+		return nil
+	}
+
+	return errors.New("timed out to get non-empty Consul leader")
+}
+
+// getNonEmptyConsulLeader makes http request call to get the registry Consul leader
+// the response of getting leader call could be an empty leader (represented by "")
+// even if the http status code is 200 when Consul is just booting up and
+// it will take a bit of time to elect the raft leader
+func (c *cmd) getNonEmptyConsulLeader() error {
+	getLeaderURL, err := c.getRegistryApiUrl(consulGetLeaderAPI)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(http.MethodGet, getLeaderURL, http.NoBody)
+	if err != nil {
+		return fmt.Errorf("Failed to prepare request for http URL: %w", err)
+	}
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return fmt.Errorf("Failed to send request for http URL: %w", err)
+	}
+
+	defer func() {
+		_ = resp.Body.Close()
+	}()
+
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return fmt.Errorf("Failed to read response body to get leader: %w", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		trimmedResp := strings.TrimSpace(string(responseBody))
+		// Consul's raft leader election process could take a bit of time
+		// before responds back with a non-empty leader
+		if len(trimmedResp) == 0 || emptyLeader == trimmedResp {
+			return errors.New("no leader yet")
+		}
+		// now we have a cluster raft leader
+		c.loggingClient.Infof("leader [%s] is elected", trimmedResp)
+		return nil
+
+	// almost unlikely for this case unless URL is incorrect
+	default:
+		return fmt.Errorf("get Consul leader request failed with status code= %d: %s", resp.StatusCode, string(responseBody))
+	}
+}
+
+func (c *cmd) getSecretStoreTokenFromFile() (string, error) {
+	trimmedFilePath := strings.TrimSpace(c.configuration.StageGate.Registry.ACL.SecretsAdminTokenPath)
+	if len(trimmedFilePath) == 0 {
+		return emptyToken, errors.New("required StageGate_Registry_SecretsAdminTokenPath from configuration is empty")
+	}
+
+	tokenFileAbsPath, err := filepath.Abs(trimmedFilePath)
+	if err != nil {
+		return emptyToken, fmt.Errorf("failed to convert tokenFile to absolute path %s: %v", trimmedFilePath, err)
+	}
+
+	// since the secretstore token is created by another service, secretstore-setup,
+	// so here we want to make sure we have the file
+	if exists := helper.CheckIfFileExists(tokenFileAbsPath); !exists {
+		return emptyToken, fmt.Errorf("secretstore token file %s not found", tokenFileAbsPath)
+	}
+
+	type SecretStoreToken struct {
+		Authentication struct {
+			Token string `json:"client_token"`
+		} `json:"auth"`
+	}
+
+	tokenJSONFile, err := os.Open(trimmedFilePath)
+	defer func() { tokenJSONFile.Close() }()
+
+	if err != nil {
+		return emptyToken, fmt.Errorf("failed to open secretstore token file: %v", err)
+	}
```

Comment:
Replace with call to github.com/edgexfoundry/go-mod-secrets/pkg/token/authtokenloader

Reply:
ok, will need to spend time on trying to understand that piece first in order to fit in here

Reply:
replaced